File: MotorControl.c
#include <stdio.h>
#include <wiringPi.h>

#define IN1 15
#define IN2 16
#define IN3 1
#define IN4 4

#define IN5 5
#define IN6 6
#define IN7 10
#define IN8 11

int Steps1 = 0;
int Direction1 = 1;
int Steps2 = 0;
int Direction2 = 1;

void setup() {
	//Serial.begin(9600);
	pinMode(IN1, OUTPUT);
	pinMode(IN2, OUTPUT);
	pinMode(IN3, OUTPUT);
	pinMode(IN4, OUTPUT);
	
	pinMode(IN5, OUTPUT);
	pinMode(IN6, OUTPUT);
	pinMode(IN7, OUTPUT);
	pinMode(IN8, OUTPUT);
}

void loop() {
	for(int i=0; i<4096; i++){
		stepper1(1);
		delayMicroseconds(800);
	}
	Direction1 = !Direction1;
	for(int i=0; i<4096; i++){
		stepper1(1);
		delayMicroseconds(800);
	}
	
	for(int i=0; i<4096; i++){
		stepper2(1);
		delayMicroseconds(800);
	}
	Direction2 = !Direction2;
	for(int i=0; i<4096; i++){
		stepper2(1);
		delayMicroseconds(800);
	}
}

void stepper1(int xw) {
	for (int x = 0; x < xw; x++) {
		switch (Steps1) {
			case 0:
				digitalWrite(IN1, LOW);
				digitalWrite(IN2, LOW);
				digitalWrite(IN3, LOW);
				digitalWrite(IN4, HIGH);
				break;
			case 1:
				digitalWrite(IN1, LOW);
				digitalWrite(IN2, LOW);
				digitalWrite(IN3, HIGH);
				digitalWrite(IN4, HIGH);
				break;
			case 2:
				digitalWrite(IN1, LOW);
				digitalWrite(IN2, LOW);
				digitalWrite(IN3, HIGH);
				digitalWrite(IN4, LOW);
				break;
			case 3:
				digitalWrite(IN1, LOW);
				digitalWrite(IN2, HIGH);
				digitalWrite(IN3, HIGH);
				digitalWrite(IN4, LOW);
				break;
			case 4:
				digitalWrite(IN1, LOW);
				digitalWrite(IN2, HIGH);
				digitalWrite(IN3, LOW);
				digitalWrite(IN4, LOW);
				break;
			case 5:
				digitalWrite(IN1, HIGH);
				digitalWrite(IN2, HIGH);
				digitalWrite(IN3, LOW);
				digitalWrite(IN4, LOW);
				break;
			case 6:
				digitalWrite(IN1, HIGH);
				digitalWrite(IN2, LOW);
				digitalWrite(IN3, LOW);
				digitalWrite(IN4, LOW);
				break;
			case 7:
				digitalWrite(IN1, HIGH);
				digitalWrite(IN2, LOW);
				digitalWrite(IN3, LOW);
				digitalWrite(IN4, HIGH);
				break;
			default:
				digitalWrite(IN1, LOW);
				digitalWrite(IN2, LOW);
				digitalWrite(IN3, LOW);
				digitalWrite(IN4, LOW);
				break;
		}
		StepOne1();
	}
}

void stepper2(int xw) {
	for (int x = 0; x < xw; x++) {
		switch (Steps2) {
			case 0:
				digitalWrite(IN5, LOW);
				digitalWrite(IN6, LOW);
				digitalWrite(IN7, LOW);
				digitalWrite(IN8, HIGH);
				break;
			case 1:
				digitalWrite(IN5, LOW);
				digitalWrite(IN6, LOW);
				digitalWrite(IN7, HIGH);
				digitalWrite(IN8, HIGH);
				break;
			case 2:
				digitalWrite(IN5, LOW);
				digitalWrite(IN6, LOW);
				digitalWrite(IN7, HIGH);
				digitalWrite(IN8, LOW);
				break;
			case 3:
				digitalWrite(IN5, LOW);
				digitalWrite(IN6, HIGH);
				digitalWrite(IN7, HIGH);
				digitalWrite(IN8, LOW);
				break;
			case 4:
				digitalWrite(IN5, LOW);
				digitalWrite(IN6, HIGH);
				digitalWrite(IN7, LOW);
				digitalWrite(IN8, LOW);
				break;
			case 5:
				digitalWrite(IN5, HIGH);
				digitalWrite(IN6, HIGH);
				digitalWrite(IN7, LOW);
				digitalWrite(IN8, LOW);
				break;
			case 6:
				digitalWrite(IN5, HIGH);
				digitalWrite(IN6, LOW);
				digitalWrite(IN7, LOW);
				digitalWrite(IN8, LOW);
				break;
			case 7:
				digitalWrite(IN5, HIGH);
				digitalWrite(IN6, LOW);
				digitalWrite(IN7, LOW);
				digitalWrite(IN8, HIGH);
				break;
			default:
				digitalWrite(IN5, LOW);
				digitalWrite(IN6, LOW);
				digitalWrite(IN7, LOW);
				digitalWrite(IN8, LOW);
				break;
		}
		StepOne2();
	}
}

void StepOne1() {
	if (Direction1 == 1) {
		Steps1++;
	}
	if (Direction1 == 0) {
		Steps1--;
	}
	if (Steps1 > 7) {
		Steps1 = 0;
	}
	if (Steps1 < 0) {
		Steps1 = 7;
	}
}

void StepOne2() {
	if (Direction2 == 1) {
		Steps2++;
	}
	if (Direction2 == 0) {
		Steps2--;
	}
	if (Steps2 > 7) {
		Steps2 = 0;
	}
	if (Steps2 < 0) {
		Steps2 = 7;
	}
}

int main(void)
{
	if(wiringPiSetup() == -1){
		//if the wiringPi initialization fails, print error message
		printf("setup wiringPi failed !");
		return 1;
	}
	setup();
	loop();
	return 0;
}

File: MotorControl.py
import wiringpi

IN1 = 15
IN2 = 16
IN3 = 1
IN4 = 4

IN5 = 5
IN6 = 6
IN7 = 10
IN8 = 11

OUTPUT = 1

caseMap = [
	[0,0,0,1],
	[0,0,1,1],
	[0,0,1,0],
	[0,1,1,0],
	[0,1,0,0],
	[1,1,0,0],
	[1,0,0,0],
	[1,0,0,1]
]

Steps1 = 0
Direction1 = True
Steps2 = 0
Direction2 = True

def setup():
	# Serial.begin(9600)
	wiringpi.wiringPiSetupGpio()
	
	wiringpi.pinMode(IN1, OUTPUT)
	wiringpi.pinMode(IN2, OUTPUT)
	wiringpi.pinMode(IN3, OUTPUT)
	wiringpi.pinMode(IN4, OUTPUT)
	
	wiringpi.pinMode(IN5, OUTPUT)
	wiringpi.pinMode(IN6, OUTPUT)
	wiringpi.pinMode(IN7, OUTPUT)
	wiringpi.pinMode(IN8, OUTPUT)


def loop():
	for i in range(4096):
		stepper1(1)
		delayMicroseconds(800)
	
	Direction1 = not Direction1
	for i in range(4096):
		stepper1(1)
		delayMicroseconds(800)
	
	
	for i in range(4096):
		stepper2(1)
		delayMicroseconds(800)
	
	Direction2 = not Direction2
	for i in range(4096):
		stepper2(1)
		delayMicroseconds(800)
	

def stepper1(xw):
	for x in range(xw):
		wiringpi.digitalWrite(IN1, caseMap[Steps1][0])
		wiringpi.digitalWrite(IN2, caseMap[Steps1][1])
		wiringpi.digitalWrite(IN3, caseMap[Steps1][2])
		wiringpi.digitalWrite(IN4, caseMap[Steps1][3])
		StepOne1()
	

def stepper2(xw):
	for x in range(xw):
		wiringpi.digitalWrite(IN5, caseMap[Steps1][0])
		wiringpi.digitalWrite(IN6, caseMap[Steps1][1])
		wiringpi.digitalWrite(IN7, caseMap[Steps1][2])
		wiringpi.digitalWrite(IN8, caseMap[Steps1][3])
		StepOne2()


def StepOne1():
	if Direction1:
		Steps1 += 1
	else:
		Steps1 -= 1
	
	if Steps1 > 7:
		Steps1 = 0
	
	if Steps1 < 0:
		Steps1 = 7


def StepOne2():
	if Direction2:
		Steps2 += 1
	else:
		Steps2 -= 1
	
	if Steps2 > 7:
		Steps2 = 0
	
	if Steps2 < 0:
		Steps2 = 7
	


def main():
	if wiringpi.wiringPiSetup() == -1:
		# if the wiringPi initialization fails, print error message
		print("setup wiringPi failed !")
		return 1
	
	setup()
	loop()
	return 0
